x/group/server/invariants.go

```diff
@@ -169,3 +181,126 @@ func groupTotalWeightInvariant(ctx sdk.Context, groupTable orm.Table, groupMembe
 	}
 	return msg, broken, err
 }
+
+func tallyVotesSumInvariant(ctx sdk.Context, proposalTable orm.AutoUInt64Table, groupAccountTable orm.PrimaryKeyTable, groupMemberByGroupIndex orm.UInt64Index, voteTable orm.PrimaryKeyTable) (string, bool, error) {
+	var msg string
+	var broken bool
+
+	var proposal group.Proposal
+	var groupAcc group.GroupAccountInfo
+	var groupMem group.GroupMember
+	var vote group.Vote
+
+	proposalIt, err := proposalTable.PrefixScan(ctx, 1, math.MaxUint64)
+	if err != nil {
+		return msg, broken, err
+	}
+	defer proposalIt.Close()
+
+	for {
+
+		totalVotingWeight := apd.New(0, 0)
+		yesVoteWeight := apd.New(0, 0)
+		noVoteWeight := apd.New(0, 0)
+		abstainVoteWeight := apd.New(0, 0)
+		vetoVoteWeight := apd.New(0, 0)
+
+		_, err := proposalIt.LoadNext(&proposal)
+		if orm.ErrIteratorDone.Is(err) {
+			break
+		}
+		address, err := sdk.AccAddressFromBech32(proposal.Address)
+		if err != nil {
+			return msg, broken, err
+		}
+		err = groupAccountTable.GetOne(ctx, address.Bytes(), &groupAcc)
+		if err != nil {
+			break
+		}
+		groupMemIt, err := groupMemberByGroupIndex.Get(ctx, groupAcc.GroupId)
+		if err != nil {
+			return msg, broken, err
+		}
+		defer groupMemIt.Close()
+
+		for {
```

Comment:
I believe you shouldn't loop through all the members but rather through all the votes for a given proposal.
Imagine that you have a proposal where not all members have voted, when you try to ` voteTable.GetOne` for a member who hasn't voted, it'll return an error

Reply:
Yeah, makes sense